omega/omega-transport/omega-transport-servicecomb/src/main/java/org/apache/servicecomb/saga/omega/transport/servicecomb/SagaProviderHandler.java

```diff
@@ -22,27 +22,35 @@
 
 import java.lang.invoke.MethodHandles;
 
+import org.apache.servicecomb.core.Handler;
+import org.apache.servicecomb.core.Invocation;
+import org.apache.servicecomb.foundation.common.utils.BeanUtils;
 import org.apache.servicecomb.saga.omega.context.OmegaContext;
+import org.apache.servicecomb.swagger.invocation.AsyncResponse;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
-import org.springframework.beans.factory.annotation.Autowired;
 
-import org.apache.servicecomb.core.Handler;
-import org.apache.servicecomb.core.Invocation;
-import org.apache.servicecomb.swagger.invocation.AsyncResponse;
+import com.google.common.annotations.VisibleForTesting;
 
 public class SagaProviderHandler implements Handler {
 
   private static final Logger LOG = LoggerFactory.getLogger(MethodHandles.lookup().lookupClass());
-  private final OmegaContext omegaContext;
 
-  public SagaProviderHandler(@Autowired(required=false) OmegaContext omegaContext) {
-    this.omegaContext = omegaContext;
-    if (omegaContext == null) {
-      LOG.info("The OmegaContext is not injected, The SagaProviderHander is disabled.");
+  private OmegaContext omegaContext;
```

Comment:
It's better to use the final object to avoid the change from outside.
We could add a new construction method which set the omegaContext to be null.

Reply:
Done